mcs/mcs/expression.cs

```diff
@@ -1692,37 +1692,70 @@ void EmitLoad (EmitContext ec)
 					ec.EmitThis ();
 
 				expr_unwrap.Emit (ec);
-			} else {
-				if (Variable?.HoistedVariant != null)
-					ec.EmitThis ();
 
-				expr.Emit (ec);
+				if (Variable.HoistedVariant != null) {
+					Variable.HoistedVariant.EmitAssignFromStack (ec);
+				} else {
+					//
+					// It's ok to have variable builder created out of order. It simplifies emit
+					// of statements like while (condition) { }
+					//
+					if (!Variable.Created)
+						Variable.CreateBuilder (ec);
 
-				// Only to make verifier happy
-				if (probe_type_expr.IsGenericParameter && TypeSpec.IsValueType (expr.Type))
-					ec.Emit (OpCodes.Box, expr.Type);
+					Variable.EmitAssign (ec);
+				}
 
-				ec.Emit (OpCodes.Isinst, probe_type_expr);
+				ec.MarkLabel (no_value_label);
```

Comment:
not visible in the diff, but you can move the `no_value_label` declaration into the `expr_unwrap` case and also get rid of a `new Label ()` allocation.

Reply:
yeah, it's struct but it looks better